public/scripts/slash-commands/SlashCommandCommonEnumsProvider.js

```diff
@@ -251,14 +252,22 @@ export const commonEnumProviders = {
      * @param {boolean} [options.allowVars=false] - Whether to add enum option for variable names
      * @returns {(executor:SlashCommandExecutor, scope:SlashCommandScope) => SlashCommandEnumValue[]}
      */
-    messages: ({ allowIdAfter = false, allowVars = false } = {}) => (_, scope) => {
+    messages: ({ allowIdAfter = false, allowVars = false } = {}) => (executor, scope) => {
+        const nameFilter = executor.namedArgumentList.find(it => it.name == 'name')?.value || '';
         return [
-            ...chat.map((message, index) => new SlashCommandEnumValue(String(index), `${message.name}: ${message.mes}`, enumTypes.number, message.is_user ? enumIcons.user : message.is_system ? enumIcons.system : enumIcons.assistant)),
+            ...chat.map((message, index) => new SlashCommandEnumValue(String(index), `${message.name}: ${message.mes}`, enumTypes.number, message.is_user ? enumIcons.user : message.is_system ? enumIcons.system : enumIcons.assistant)).filter(value => !nameFilter || value.description.startsWith(`${nameFilter}:`)),
             ...allowIdAfter ? [new SlashCommandEnumValue(String(chat.length), '>> After Last Message >>', enumTypes.enum, '➕')] : [],
-            ...allowVars ? commonEnumProviders.variables('all')(_, scope) : [],
+            ...allowVars ? commonEnumProviders.variables('all')(executor, scope) : [],
         ];
     },
 
+    /**
+     * All names used in the current chat.
+     *
+     * @returns {SlashCommandEnumValue[]}
+     */
+    messageNames: () => chat.map((message) => message.name).filter(onlyUnique).sort(Intl.Collator().compare).map(name => new SlashCommandEnumValue(name)),
```

Comment:
Suggestion: Different icon for char and persona names (maybe even system) here.
If we are already on it.

Also, what the heck is `Intl.Collator().compare`?
Shouldn't we just finally make a re-usable actual compare version based on `compareIgnoreCaseAndAccents` that is already used for searching in many other places?